pkg/destroy/powervs/serviceinstance.go

```diff
@@ -0,0 +1,314 @@
+		moreData              = true
+		nextURL         *string
+	)
+
+	resourceGroupID, err = o.convertResourceGroupNameToID(o.resourceGroupID)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to convert resourceGroupID")
+	}
+	o.Logger.Debugf("listServiceInstances: converted %v to %v", o.resourceGroupID, resourceGroupID)
+
+	options = o.controllerSvc.NewListResourceInstancesOptions()
+	// options.SetType("resource_instance")
+	options.SetResourceGroupID(resourceGroupID)
+	options.SetResourceID(virtualServerResourceID)
+	options.SetLimit(perPage)
+
+	result := []cloudResource{}
+
+	for moreData {
+		if options.Start != nil {
+			o.Logger.Debugf("listServiceInstances: options = %+v, options.Limit = %v, options.Start = %v, options.ResourceGroupID = %v", options, *options.Limit, *options.Start, *options.ResourceGroupID)
+		} else {
+			o.Logger.Debugf("listServiceInstances: options = %+v, options.Limit = %v, options.ResourceGroupID = %v", options, *options.Limit, *options.ResourceGroupID)
+		}
+
+		resources, _, err = o.controllerSvc.ListResourceInstancesWithContext(ctx, options)
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to list resource instances")
+		}
+
+		o.Logger.Debugf("listServiceInstances: resources.RowsCount = %v", *resources.RowsCount)
+
+		for _, resource := range resources.Resources {
+			var (
+				getResourceOptions *resourcecontrollerv2.GetResourceInstanceOptions
+				resourceInstance   *resourcecontrollerv2.ResourceInstance
+				response           *core.DetailedResponse
+			)
+
+			o.Logger.Debugf("listServiceInstances: resource.Name = %s", *resource.Name)
+
+			getResourceOptions = o.controllerSvc.NewGetResourceInstanceOptions(*resource.ID)
+
+			resourceInstance, response, err = o.controllerSvc.GetResourceInstance(getResourceOptions)
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed to get instance: %s", response)
+			}
+			if response != nil && response.StatusCode == gohttp.StatusNotFound {
+				o.Logger.Debugf("listServiceInstances: gohttp.StatusNotFound")
+				continue
+			} else if response != nil && response.StatusCode == gohttp.StatusInternalServerError {
+				o.Logger.Debugf("listServiceInstances: gohttp.StatusInternalServerError")
+				continue
+			}
+
+			if resourceInstance.Type == nil {
+				o.Logger.Debugf("listServiceInstances: type: nil")
+			} else {
+				o.Logger.Debugf("listServiceInstances: type: %v", *resourceInstance.Type)
+			}
+			if resourceInstance.SubType == nil {
+				o.Logger.Debugf("listServiceInstances: sub type: nil")
+			} else {
+				o.Logger.Debugf("listServiceInstances: sub type: %v", *resourceInstance.SubType)
+			}
+			if resourceInstance.ResourcePlanID == nil {
+				o.Logger.Debugf("listServiceInstances: plan: nil")
+			} else {
+				o.Logger.Debugf("listServiceInstances: plan: %v", *resourceInstance.ResourcePlanID)
+			}
+			o.Logger.Debugf("listServiceInstances: TEST1 = %v", resourceInstance.Type != nil && *resourceInstance.Type == "service_instance")
```

Comment:
Can these debug statements be removed prior to merge since they're adding up quite a large amount of command-line output?

Reply:
I removed some of them.  Since it is debug output, I am not worried about taking up a lot of output with potentially helpful debugging output.